chain/types/ethtypes/eth_transactions.go

```diff
@@ -227,347 +184,90 @@ func (tx *EthTxArgs) ToSignedMessage() (*types.SignedMessage, error) {
 	}, nil
 }
 
-func (tx *EthTxArgs) HashedOriginalRlpMsg() ([]byte, error) {
-	msg, err := tx.ToRlpUnsignedMsg()
-	if err != nil {
-		return nil, err
-	}
-
-	hasher := sha3.NewLegacyKeccak256()
-	hasher.Write(msg)
-	hash := hasher.Sum(nil)
-	return hash, nil
-}
-
-func (tx *EthTxArgs) ToRlpUnsignedMsg() ([]byte, error) {
-	packed, err := tx.packTxFields()
-	if err != nil {
-		return nil, err
-	}
-
-	encoded, err := EncodeRLP(packed)
-	if err != nil {
-		return nil, err
-	}
-	return append([]byte{0x02}, encoded...), nil
-}
-
-func (tx *EthTx) ToEthTxArgs() EthTxArgs {
-	return EthTxArgs{
-		ChainID:              int(tx.ChainID),
-		Nonce:                int(tx.Nonce),
-		To:                   tx.To,
-		Value:                big.Int(tx.Value),
-		MaxFeePerGas:         big.Int(tx.MaxFeePerGas),
-		MaxPriorityFeePerGas: big.Int(tx.MaxPriorityFeePerGas),
-		GasLimit:             int(tx.Gas),
-		Input:                tx.Input,
-		V:                    big.Int(tx.V),
-		R:                    big.Int(tx.R),
-		S:                    big.Int(tx.S),
+func ParseEthTransaction(data []byte) (EthTransaction, error) {
+	if len(data) == 0 {
+		return nil, fmt.Errorf("empty data")
 	}
-}
-
-func (tx *EthTx) TxHash() (EthHash, error) {
-	ethTxArgs := tx.ToEthTxArgs()
-	return (&ethTxArgs).TxHash()
-}
 
-func (tx *EthTxArgs) TxHash() (EthHash, error) {
-	rlp, err := tx.ToRlpSignedMsg()
-	if err != nil {
-		return EmptyEthHash, err
+	if data[0] > 0x7f {
```

Comment:
This is found in 2 places in this diff so should probably be documented in both places. I think the best resource might be: https://eips.ethereum.org/EIPS/eip-2718#backwards-compatibility, and then specifically EIP-1559 builds on that by using `0x02` as its prefix @ https://eips.ethereum.org/EIPS/eip-1559#abstract

Also interesting is that `[0xc0, 0xfe]` is specifically called out as the logical range for legacy transactions. I don't know if we need to add that check because out of that range should either be picked up by bad sig or bad RLP encoding, but it's an option apparently.